plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -331,6 +332,15 @@ public void testSqlServerNchar()
                 .execute(getQueryRunner(), sqlServerCreateAndInsert("test_sqlserver_char"));
     }
 
+    @Test
+    public void testSqlServerUniqueIdentifier()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("uniqueidentifier", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uniqueidentifier", "'d5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
```

Comment:
Could you add below round trips? 
* `00000000-0000-0000-0000-000000000000`
* `ffffffff-ffff-ffff-ffff-ffffffffffff`

Also, SQL Server accepts invalid uuid string if I understand correctly, e.g. `0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong`. It would be nice to test such values as well.

Reply:
Okay

Reply:
As I tested with SQL server,
* `00000000-0000-0000-0000-000000000000` should succeed
* `ffffffff-ffff-ffff-ffff-ffffffffffff` should succeed
* `0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong` the process flow is
  * Check if the provided string has size at least `36` if not query fails
  * Truncate provided string to size `36` then validate if it forms a UUID

Should also add a case to insert `0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong` and expect `0E984725-C51C-4BF4-9960-E1C80E27ABA0` when reading from Trino.

Reply:
IMO we should fail such INSERTs when done from Trino. It's data corruption.

Reply:
@hashhar I agree, but IIUC in these tests we are not going to test the cases that will fail to insert right? The invalid UUID will not be accepted by SQL Server or Trino. The case of `0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong` can be tested with `sqlServerCreateAndInsert`

Reply:
things which fail (negative tests) are also useful - especially in this case since we know that SQL Server's logic truncates value and tries again.

See existing cases like various `testUnsupportedXXX` methods in type mapping tests for other connectors.